### DISCUSSION TITLE
How to set value of DP

### DISCUSSION THREAD

**Discussion by Brazen00** [make-all/tuya-local#975] [Q&A] [answered]
This is a great component. I'm playing around with adding a config for a tap timer I have, and am wondering if there is a way of setting the value of a DP from within HA, similar to the `localtuya.set_dp` service?

**Comment by make-all** [comment, marked as answer]
This integration is intended to be high level for end users, and does not implement low level services like that.  To set a value to a dp, it needs to be mapped to a settable attribute of an entity in Home Assistant.

**Reply by JackJackattack4** [reply]
Tuya Local integration works for setting value controlling heat pumps temperature in my case. But even if properly mapped, I have trouble to set DP value in one case.

A DP called "Set_value" is used by the Tuya device to receive a new value as instruction. In my situation, I mapped this "set_value" DP is an attribute of an entity I have been successful so far to be change the numeric value of the corresponding attribute in HA. However, changing the numeric value of the attribute in HA does not change the "set_value" value in the Tuya device. Actually, in HA, after few seconds, the attribute reverts back to the previous value it had before I tempered with it via the python script I use. I'm missing a "eat it" action from HA, to get the Tuya device to receive the instruction to take the new number.

In Tuya IoT platform, I have to click "send instruction" button in the "device debugging" page to get the new set_value instruction in the tuya device. In the Tuya app, after changing the "set_value" value, I must click confirm for the number to be taken by the Tuya device.

Any idea on how to tell the Tuya device from HA to "eat it"?

**Reply by JackJackattack4** [reply]
I thought maybe this has to do with the format of the input value. The DP "set_value" requires a integer as per Tuya IoT Platform. But with this code, the input number should be an integer, the Tuya Device still rejects it.

service: python_script.hass_entities
data:
  action: set_attributes
  entity_id: climate.deakon_hp_chaud_froid
  attributes:
    - Set_Value : "{{states('input_number.defrosting_period' )|int}}"

**Reply by make-all** [reply]
Home Assistant does not provide any API to set arbitrary attributes on a climate entity.  I don't know how your script is managing to set the values, but most likely it is only affecting HA internal state, not getting through to the device, as there is no API for that.  As mentioned above, you need to map the attribute to an appropriate settable attribute on an entity, such as a number entity for arbitrary numeric values.
